src/Compilers/CSharp/Test/Symbol/Compilation/CompilationAPITests.cs

```diff
@@ -2053,6 +2053,30 @@ public void PreviousSubmissionWithError()
             Assert.Throws<InvalidOperationException>(() => CreateSubmission("a + 1", previous: s0));
         }
 
+        [Fact]
+        public void CreateTupleTypeSymbol()
+        {
+            var comp = CSharpCompilation.Create("test", references: new[] { MscorlibRef } );
+            ITypeSymbol intType = comp.GetSpecialType(SpecialType.System_Int32);
+            ITypeSymbol stringType = comp.GetSpecialType(SpecialType.System_String);
+
+            Assert.Throws<ArgumentNullException>(() => comp.CreateTupleTypeSymbol(default(ImmutableArray<ITypeSymbol>), default(ImmutableArray<string>)));
+
+            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol>.Empty, default(ImmutableArray<string>)));
+            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(new[] { intType }.AsImmutable(), default(ImmutableArray<string>)));
```

Comment:
My focus here is to test the new code. I'm hesitant to recreate a new matrix for every entry point that hits the core logic of the symbol, but I'm open a reasonable subset. 
I'll split the test to make it easier to follow, add some success cases, add VB types, and add long tuples.